solr/core/src/java/org/apache/solr/search/SurroundQParserPlugin.java

```diff
@@ -40,6 +40,35 @@
 public class SurroundQParserPlugin extends QParserPlugin {
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
   public static final String NAME = "surround";
+  private static final String MAX_BASIC_QUERIES_SYSTEM_PROP = "solr.absoluteMaxBasicQueries";
+  private static final int ABSOLUTE_MAX_BASIC_QUERIES = readAbsolutionMaxBasicQueries();
+
+  private static int readAbsolutionMaxBasicQueries() {
+    String maxBasicQueriesSystemProp = System.getProperty(MAX_BASIC_QUERIES_SYSTEM_PROP);
+    if (maxBasicQueriesSystemProp != null) {
+      try {
+        int absoluteMaxBasicQueries = Integer.parseInt(maxBasicQueriesSystemProp);
+        if (absoluteMaxBasicQueries > 0) {
+          log.info(
+              "maxBasicQueries with system property {} with value {}",
+              MAX_BASIC_QUERIES_SYSTEM_PROP,
+              maxBasicQueriesSystemProp);
+          return absoluteMaxBasicQueries;
+        } else {
+          log.info(
+              "Ignoring system property {} value {} since it is non-positive",
+              MAX_BASIC_QUERIES_SYSTEM_PROP,
+              maxBasicQueriesSystemProp);
+        }
+      } catch (NumberFormatException e) {
```

Comment:
Maybe just set if positive value otherwise ignore it.

Reply:
This code does something very similar?
if `solr.absoluteMaxBasicQueries` is positive, set it (by returning the value)
otherwise, ignore it (by returning -1, which indicates no ABSOLUTE_MAX_BASIC_QUERIES) ?